### DISCUSSION TITLE
Order of collection with `orderBy` is not correct after loading after clearing EM

### DISCUSSION THREAD

**Discussion by TiesB** [mikro-orm/mikro-orm#5999] [Q&A] [answered]
### Describe the bug

When a property that is included in a `orderBy` setting is changed (and therefore the intended ordering of the collection changes), and afterwards the EM is cleared, and the collection retrieved through `loadItems`, the change in collection ordering is not represented. It seems to us that after using `loadItems` (especially after clearing the EM), the order of the collection should be "recalculated".

We encoutered this in testing. Obviously, we would not use such a pattern in production code.

### Reproduction

https://github.com/TiesB/mikroorm-order-repro

### What driver are you using?

None

### MikroORM version

6.3.9-dev.0

### Node.js version

Node.js 20.16.0, TypesScript 5.5.4

### Operating system

Ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment, marked as answer]
Your `author.books` is still a populated collection, just because you clear the EM that was used to load it doesn't change the fact that your collection is already initialized, hence the `loadItems` call is a no-op.

**Reply by TiesB** [reply]
D'oh, that makes a lot of sense... Thanks!
